### PR TITLE
Adding tests for the Transactions in Solidity

### PR DESCRIPTION
**Pull request by stefan-stefanooov**
**Description**:
Add support for transaction properties

**Related issue(s)**:
https://github.com/hashgraph/hedera-smart-contracts/issues/431

Fixes #

**Notes for reviewer**:

### CODE REVIEW

File: contracts/solidity/transaction/Transaction.sol [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    function getMessageValue() public payable {
+        emit MsgValue(msg.value);
+    }
+
+    function getGasPrice() public view returns (uint256) {
+        return tx.gasprice;
+    }
+
+    function getTxOriginFromSecondary() public view returns (address) {
+        address secondaryOrigin = secondaryContract.getTxOrigin();
```
Comment (natanasow) [inline contracts/solidity/transaction/Transaction.sol:46]: The local `secondaryOrigin` variable is redundant. The method could be simplified to 
```solidity
function getTxOriginFromSecondary() public view returns (address) {
    return secondaryContract.getTxOrigin();
}
```

File: test/solidity/transaction/transaction.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    const signers = await ethers.getSigners()
+    wallet = signers[0];
+    senderWalletAddr = await wallet.getAddress()
+  })
+
+  it('gasleft() returns (uint256): remaining gas', async function () {
+    const gasLeft = await contractTr.checkGasleft()
+    
+    expect(gasLeft).to.exist
+    expect(ethers.BigNumber.isBigNumber(gasLeft)).to.be.true
```
Comment (Nana-EC) [inline test/solidity/transaction/transaction.js:53]: Would  be good here to pass in a gas amount and check that the `0 < gasLeft < gasSent`

File: test/solidity/transaction/transaction.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    
+    // to compare with the value sent, we need to convert to tinybar
+    expect(amount.mul('10000000000')).to.equal(valueToSend)
+  })
+
+  it('tx.gasprice (uint): gas price of the transaction', async function () {
+    const gasPrice = await contractTr.getGasPrice()
+
+    expect(gasPrice).to.exist
+    expect(ethers.BigNumber.isBigNumber(gasPrice)).to.be.true
```
Comment (Nana-EC) [inline test/solidity/transaction/transaction.js:106]: Add greater than 0 check here and any other numbers.
0 gets used often as a default and can signal a bug

File: contracts/solidity/transaction/Secondary.sol [hunk 4]
```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+contract Secondary {
```
Comment (Nana-EC) [inline contracts/solidity/transaction/Secondary.sol:4]: nit: Looking for less general names. What about `MessageFrameAddresses` or `MessageFrameAccounts`?
Ideas?

**Review by Nana-EC — CHANGES_REQUESTED** [review]
nice, some suggestions

File: test/solidity/transaction/transaction.js (hunk 3 above)
Comment (stefan-stefanooov) [reply test/solidity/transaction/transaction.js:106]: Thanks, good to know.
I added `greater than 0` check. 
I am not sure I understand the second part of the comment `...and any other numbers.`

File: contracts/solidity/transaction/Secondary.sol (hunk 4 above)
Comment (stefan-stefanooov) [reply contracts/solidity/transaction/Secondary.sol:4]: `MessageFrameAddresses` sounds good to me.
Thanks ... got a bit lazy with the naming I guess.

File: test/solidity/transaction/transaction.js (hunk 2 above)
Comment (stefan-stefanooov) [reply test/solidity/transaction/transaction.js:53]: Done.

File: test/solidity/transaction/transaction.js (hunk 3 above)
Comment (Nana-EC) [reply test/solidity/transaction/transaction.js]: Yeah i just meant any other values you may be checking in EVM responses you might want to confirm non zero also

**Review by Nana-EC — APPROVED** [review]
LGTM

File: test/solidity/transaction/transaction.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+ * limitations under the License.
+ *
+ */
+
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+const Constants = require('../../constants')
+const Utils = require('../../hts-precompile/utils')
+
+describe('PrngSystemContract tests', function () {
```
Comment (quiet-node) [inline test/solidity/transaction/transaction.js:26]: Shouldn't this be 'transaction tests' instead?

Comment (Nana-EC) [reply test/solidity/transaction/transaction.js:26]: @quiet-node please fix this and I can sign off

**Review by Nana-EC — APPROVED** [review]
LGTM

**Comment by stefan-stefanooov** [comment]
Thanks for the help here @quiet-node ;)

**Pull request merged**